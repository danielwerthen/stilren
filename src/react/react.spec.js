import React from 'react';
import ReactDOM from 'react-dom/server';
import { Server } from 'styletron-engine-atomic';
import createFactory, { Provider } from './index';

const { createElement } = React;

const factory = createFactory({
  createElement,
  breakpoints: {
    alpha: '(min-width: 100px)',
    beta: '(max-width: 800px)',
  },
  pseudos: ['hover', 'focus'],
});

function render(Component, props, ...children) {
  const styletron = new Server();
  expect(
    ReactDOM.renderToString(
      createElement(
        Provider,
        { value: styletron },
        createElement(Component, props, ...children)
      )
    )
  ).toMatchSnapshot();
  expect(styletron.getCss()).toMatchSnapshot();
}

describe('React createFactory', () => {
  it('should render stuff', () => {
    class MyStyle {
      static defaultStyle() {
        return {
          fontSize: '10px',
          color: 'black',
        };
      }
    }
    const MyComponent = factory('div', new MyStyle());
    render(
      MyComponent,
      {
        $fontSize: '12px',
        $alphaFontSize: '22px',
      },
      'foobar'
    );
  });
  it('should render animations', () => {
    class MyStyle {
      static defaultStyle() {
        return {
          animationName: {
            from: {
              opacity: 0,
            },
            to: {
              opacity: 0.5,
            },
          },
          fontFamily: {
            src: 'url("/fonts") format("woff2"), url("/font2") format("woff")',
          },
        };
      }
    }
    const MyComponent = factory('div', new MyStyle());
    render(MyComponent, null, 'foobar');
  });
});
