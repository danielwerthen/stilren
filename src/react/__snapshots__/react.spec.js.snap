// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React createFactory should render animations 1`] = `"<div class=\\"ae af\\">foobar</div>"`;

exports[`React createFactory should render animations 2`] = `"@font-face{font-family:ae;src:url(\\"/fonts\\") format(\\"woff2\\"), url(\\"/font2\\") format(\\"woff\\")}.ae{-webkit-animation-name:ae;animation-name:ae}.af{font-family:ae}@keyframes ae{from{opacity:0}to{opacity:0.5}}"`;

exports[`React createFactory should render stuff 1`] = `"<div class=\\"ae af ag\\">foobar</div>"`;

exports[`React createFactory should render stuff 2`] = `".ae{font-size:12px}.af{color:black}@media (min-width: 100px){.ag{font-size:22px}}"`;
